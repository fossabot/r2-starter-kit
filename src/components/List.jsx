import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export default function List({ data }) {
    return (
        <div className="ts divided items">
            { data.map((val, key) => {
                return (
                    <div key={`item${key}`} className="item">
                        <div className="image">
                            <img src={val.img} />
                        </div>
                        <div className="content">
                            <Link to={val.link} className="header">{val.title}</Link>
                            <div className="description">
                                {val.desc}
                            </div>
                        </div>
                    </div>
                );
            })}
        </div>
    );
}

List.defaultProps = {
    data: [{
        id: 0,
        title: 'Title',
        link: '#',
        desc: 'Content',
        img: 'http://placehold.it/320x180',
    }]
};

List.propTypes = {
    data: PropTypes.arrayOf(PropTypes.shape({
        img: PropTypes.string,
        link: PropTypes.string,
        title: PropTypes.string,
        desc: PropTypes.string
    }))
};
