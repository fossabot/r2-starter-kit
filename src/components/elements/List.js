import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import FadeOutImage from 'components/elements/FadeOutImage';

export default class List extends PureComponent {
  static propTypes = {
    data: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string,
        link: PropTypes.string,
        desc: PropTypes.string,
        imgUrl: PropTypes.string,
        imgPlaceholder: PropTypes.string,
      }),
    ),
  };
  static defaultProps = {
    data: [],
  };
  render() {
    const { data } = this.props;
    return (
      <div className="ts divided items">
        {data.map((val, key) => {
          return (
            <div key={`item${key}`} className="item">
              <div className="image">
                <FadeOutImage
                  url={val.imgUrl}
                  placeholder={val.imgPlaceholder}
                />
              </div>
              <div className="content">
                <Link to={val.link} className="header">
                  {val.title}
                </Link>
                <div className="description">{val.desc}</div>
              </div>
            </div>
          );
        })}
      </div>
    );
  }
}
