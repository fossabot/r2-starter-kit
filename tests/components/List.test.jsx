import React from 'react';
import { shallow } from 'enzyme';
import List from 'components/List';

describe('<List />', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<List />);
  });

  it('should render initial component', () => {
    wrapper.setProps({
      data: [{
        id: 0,
        title: 'Title',
        link: '#',
        desc: 'Content',
        img: 'https://placehold.it/320x180',
      }]
    });
    expect(wrapper).toMatchSnapshot();
  });
});
