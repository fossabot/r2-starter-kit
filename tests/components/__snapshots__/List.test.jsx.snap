// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should render initial component 1`] = `
<div
  className="ts divided items"
>
  <div
    className="item"
    key="item0"
  >
    <div
      className="image"
      style={
        Object {
          "width": "30%",
        }
      }
    >
      <div
        style={
          Object {
            "background": "url('https://placehold.it/320x180') no-repeat center center",
            "backgroundSize": "cover",
            "display": "block",
            "paddingBottom": "56.25%",
            "width": "100%",
          }
        }
      />
    </div>
    <div
      className="content"
    >
      <Link
        className="header"
        replace={false}
        to="#"
      >
        Title
      </Link>
      <div
        className="description"
      >
        Content
      </div>
    </div>
  </div>
</div>
`;
